src/rpc/state_api.rs

```diff
@@ -200,13 +200,17 @@ pub(in crate::rpc) async fn state_fetch_root<DB: Blockstore + Sync + Send + 'sta
 
     let (car_tx, car_handle) = if let Some(save_to_file) = save_to_file {
         let (car_tx, car_rx) = flume::bounded(100);
-        let header = CarHeader::from(vec![root_cid]);
+        let roots = vec![root_cid];
         let file = tokio::fs::File::create(save_to_file).await?;
 
         let car_handle = tokio::spawn(async move {
-            let mut file = file.compat();
-            let mut stream = car_rx.stream();
-            header.write_stream_async(&mut file, &mut stream).await
+            car_rx
+                .stream()
+                .forward(
+                    CarWriter::new_carv1(roots, file)
```

Comment:
Change `CarWriter::new_carv1` to use `io::Error`. You can convert the error from `to_vec` to an io error.